src/mds/CDir.cc

```diff
@@ -3794,7 +3794,7 @@ bool CDir::should_merge() const
       return false;
   }
 
-  return (int)get_frag_size() < g_conf()->mds_bal_merge_size;
+  return ((int)get_frag_size() + (int)get_num_snap_items()) < g_conf()->mds_bal_merge_size;
```

Comment:
nit:
if should_split() tests for frag_size + num_snap_items **greater than** mds_bal_split_size
shouldn't should_merge() test for frag_size + num_snap_items **less than or equal to** mds_bal_split_size ?

or is the fuzziness about the **equality** intentional ?

Reply:
Its been this ever since. Maybe it was just overlooked-, IDK. We could probably adjust this bit in this change. I'll do that.